chain/jsonrpc/src/lib.rs

```diff
@@ -236,6 +251,138 @@ impl JsonRpcHandler {
+                        d[i] = vec[i];
+                    }
+
+                    let pk = PublicKey::ED25519(ED25519PublicKey(d));
+                    let id = pk.into();
+                    let addr = query.address.parse().map_err(|_| {
+                        RpcError::new_internal_error(
+                            None,
+                            format!("Failed to parse address: {:?}", query.address),
+                        )
+                    })?;
+
+                    let peer_info = PeerInfo::new(id, addr);
+                    let result = self
+                        .peer_manager_addr
+                        .clone()
+                        .send(OutboundTcpConnect { peer_info })
+                        .await?;
+                    Some(
+                        serde_json::to_value(result)
+                            .map_err(|err| RpcError::serialization_error(err.to_string())),
+                    )
+                }
+                "adv_accept_connection_from" => {
+                    let req =
+                        near_jsonrpc_primitives::types::requests::AcceptConnectionFromRequest::parse(
+                            params,
+                        )?;
+                    let address = req.address.parse().map_err(|_| {
+                        RpcError::new_internal_error(
+                            None,
+                            format!("Failed to parse address: {:?}", req.address),
+                        )
+                    })?;
+                    let result = self
+                        .peer_manager_addr
+                        .clone()
+                        .send(AcceptConnectionFrom { addr: address })
+                        .await?;
+                    Some(
+                        serde_json::to_value(result)
+                            .map_err(|err| RpcError::serialization_error(err.to_string())),
+                    )
+                }
+                "adv_get_peer_id" => {
+                    let response = self.peer_manager_addr.send(GetPeerId {}).await?;
+                    Some(
+                        serde_json::to_value(response)
+                            .map_err(|err| RpcError::serialization_error(err.to_string())),
+                    )
+                }
+                "adv_get_peer_list" => {
+                    let result = self.peer_manager_addr.send(PeersRequest {}).await?;
+                    Some(
+                        serde_json::to_value(result)
+                            .map_err(|err| RpcError::serialization_error(err.to_string())),
+                    )
+                }
+                "adv_get_network_stats" => {
+                    let result = self.peer_manager_addr.send(GetNetworkStats {}).await?;
+                    Some(
+                        serde_json::to_value(result)
+                            .map_err(|err| RpcError::serialization_error(err.to_string())),
+                    )
+                }
```

Comment:
Do these have to be under the adversarial flag? It feels to be that they could be useful in general.

Reply:
They don't have to be under adversarial. My plan is to combine them into one query 
`struct QueryRpc { 
    peer_id: Optional<bool>,
    get_peer_list: Optional<bool>,
    get_routing_table: Optional<bool>,
}`

`struct QueryRpcResponse { 
    peer_id: Optional<PeerId>,
    get_peer_list: Optional<Vec<PeerInfo>>,
    get_routing_table: Optional<Vec<Edge>>,
}`

I'll add a few other things here as well. This way I can avoid adding bunch of new rpc queries.
@bowenwang1996 What do you think?

Reply:
I think that is fine. @frol what do you think?